### ISSUE TITLE
Logger override bind scope of logger object to undefined

### ISSUE THREAD

**Issue by cojack** [nestjs/nest#435]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

```typescript
import {LoggerService} from '@nestjs/common';
import * as winston from 'winston';
import {DateTime} from 'luxon';


export class AppLogger implements LoggerService {
	private logger;

	constructor() {
		this.logger = new winston.Logger({
			transports: [
				new winston.transports.Console({
					label: 'Help me Im blond',
					timestamp: () => DateTime.local().toString(),
					formatter: options => `${options.timestamp()} [${options.level.toUpperCase()}] ${options.label} - ${options.message}`
				})
			]
		});
	}

	log(message: string) {
		debugger;
		this.logger.log(message);
	}
	error(message: string, trace: string) {
		this.logger.error(message);
	}
	warn(message: string) {
		this.logger.warn(message);
	}

	info(message: string) {
		this.logger.info(message);
	}
}
```

`this` is undefined

Btw, @kamilmysliwiec you're using different `interface` for passing argument, that you're expecting that Logger should have, because you're passing arguments like that:

```typescript
    log(message) {
        const { logger } = Logger;
        logger.log(message, this.context, this.isTimeDiffEnabled);
    }
    error(message, trace = '') {
        const { logger } = Logger;
        logger.error(message, trace, this.context, this.isTimeDiffEnabled);
    }
    warn(message) {
        const { logger } = Logger;
        logger.warn(message, this.context, this.isTimeDiffEnabled);
    }
```

Hmmm?

## Expected behavior

`this` should be an instance of `AppLogger`

## Minimal reproduction of the problem with instructions

Create an AppLogger like mine above and add it to the factory:

```typescript
		app = await NestFactory.create(AppModule, {
			logger: new AppLogger()
		});
```

## What is the motivation / use case for changing the behavior?

I would like to have custom logger

## Environment

<pre><code>
    "@nestjs/common": "^4.6.3",
    "@nestjs/core": "^4.6.3",
    "@nestjs/microservices": "^4.6.1",
    "@nestjs/testing": "^4.6.1",
    "@nestjs/websockets": "^4.5.8"

 
For Tooling issues:
- Node version: 8.9.4  
- Platform:  Linux

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Thanks for reporting!
```
Btw, @kamilmysliwiec you're using different interface for passing argument, that you're expecting that Logger should have, because you're passing arguments like that:
```
Actually, both `context` and `isTimeDiffEnabled` (mainly last one) aren't a part of public API, they're just needed by default logger.

**Comment by kamilmysliwiec** [comment]
Fixed in `v4.6.4` 🙂

**Comment by cojack** [comment]
@kamilmysliwiec but context would be nice ;)
